test/e2e/gateway_api_test.go

```diff
@@ -423,6 +428,284 @@ func testGatewayAPIRBAC(t *testing.T) {
+			for _, gateway := range tc.createGateways {
+				createdGateway, err := createGatewayWithListeners(gatewayClass, gateway.gatewayName, gateway.namespace, []struct {
+					Name     string
+					Hostname *string
+				}{
+					{Name: "http", Hostname: gateway.hostname},
+				})
+				gateways = append(gateways, createdGateway)
+				if err != nil {
+					t.Fatalf("failed to create gateway %s: %v", gateway.gatewayName, err)
+				}
+			}
+
+			t.Cleanup(func() {
+				for _, gateway := range gateways {
+					if err := kclient.Delete(context.TODO(), gateway); err != nil {
+						if errors.IsNotFound(err) {
+							continue
+						}
+						t.Fatalf("failed to delete gateway %q: %v", gateway.Name, err)
+					}
+				}
+			})
+
+			for _, gateway := range gateways {
+				gateway, err := assertGatewaySuccessful(t, operatorcontroller.DefaultOperandNamespace, gateway.Name)
+				if err != nil {
+					t.Fatalf("failed to create %s gateway: %v", gateway.Name, err)
+				}
+			}
+
+			if err := assertExpectedDNSRecords(t, tc.expectedDNSRecords); err != nil {
+				t.Fatalf("DNSRecord expectations not met: %v", err)
+			}
+
+			t.Logf("Check if gateways listener conditions match expected state.")
+			for _, gateway := range gateways {
+				if err := waitForGatewayListenerCondition(t, types.NamespacedName{Namespace: gateway.Namespace, Name: gateway.Name}, "http", tc.expectedListenerConditions...); err != nil {
+					t.Fatalf("did not get expected listener condition: %v", err)
+				} else {
+					t.Logf("gateway: %s listener conditions match expected state.", gateway.Name)
+				}
+			}
+		})
+	}
+
+}
+
+func testGatewayAPIDNSListenerUpdate(t *testing.T) {
+	gatewayClass, err := createGatewayClass(operatorcontroller.OpenShiftDefaultGatewayClassName, operatorcontroller.OpenShiftGatewayClassControllerName)
+	if err != nil {
+		t.Fatalf("failed to create gatewayclass: %v", err)
+	}
+
+	domain := "gws." + dnsConfig.Spec.BaseDomain
+
+	gateway, err := createGatewayWithListeners(gatewayClass, "test-gateway-update", operatorcontroller.DefaultOperandNamespace, []struct {
+		Name     string
+		Hostname *string
+	}{
+		{Name: "http-listener1", Hostname: ptr.To("foo." + domain)},
+		{Name: "http-listener2", Hostname: ptr.To("bar." + domain)},
+	})
+
+	t.Cleanup(func() {
+		gateway := gatewayapiv1.Gateway{ObjectMeta: metav1.ObjectMeta{Name: "test-gateway-update", Namespace: operatorcontroller.DefaultOperandNamespace}}
+		if err := kclient.Delete(context.TODO(), &gateway); err != nil {
+			if errors.IsNotFound(err) {
+				return
+			}
+			t.Errorf("failed to delete gateway %q: %v", gateway.Name, err)
+		}
+	})
+
+	if err != nil {
+		t.Fatalf("failed to create gateway with multiple listeners: %v", err)
+	}
+
+	gateway, err = assertGatewaySuccessful(t, operatorcontroller.DefaultOperandNamespace, "test-gateway-update")
+	if err != nil {
+		t.Fatalf("failed to create the test gateway: %v", err)
+	}
+
+	t.Logf("Created gateway %s with hostname %s", gateway.Name, "foo."+domain)
```

Comment:
This message should be logged after creation. At this point we ensured the gateway is accepted, so the message should say so.